.github/workflows/perform-release.yml

```diff
@@ -0,0 +1,142 @@
+          
+          echo "[DEBUG] Current GITHUB_OUTPUT: '$(cat $GITHUB_OUTPUT)'"
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: "Checkout Repository"
+        uses: actions/checkout@v4
+
+      - name: "Check Whether Code PR Can Be Merged"
+        if: ${{ inputs.skip-pr-merge != 'true' }}
+        uses: ./.github/actions/pr-is-mergeable
+        with:
+          pr-ref: ${{ steps.determine-branch-names.outputs.CODE_BRANCH_NAME }}
+          excluded-check-runs: |
+            {
+              \"Continuous Integration\": [\"Run BlackDuck Scan\", \"Run Security Rating\"], 
+              \"dependabot merger\": [] 
+            }
+
+      - name: "Check Code Release Commit Continuous Integration"
+        if: ${{ inputs.skip-pr-merge != 'true' }}
+        uses: ./.github/actions/workflow-succeeded
+        with:
+          workflow: "Continuous Integration"
+          sha: ${{ steps.determine-branch-names.outputs.RELEASE_COMMIT }}
+
+  release:
+    name: "Release"
+    needs: [ prerequisites ]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # needed to modify the release draft
+      pull-requests: write # needed to merge the release PR
+    steps:
+      - name: "Setup java"
+        uses: actions/setup-java@v4
+        with:
+          distribution: "sapmachine"
+          java-version: ${{ env.JAVA_VERSION }}
+          server-id: ossrh
+          server-username: MAVEN_CENTRAL_USER # env variable for username in deploy
+          server-password: MAVEN_CENTRAL_PASSWORD # env variable for token in deploy
+
+      - name: "Download Release Asset"
+        id: download-asset
+        run: |
+          gh release download ${{ needs.prerequisites.outputs.release-tag }} --dir ./ --repo "${{ github.repository }}"
+          # x=extract v=verbose z=decompress f=file C=destination directory
+          tar -xvzf release-artifacts.tar.gz -C .
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: "Import GPG Key"
+        run: |
+          echo "${{ secrets.PGP_PRIVATE_KEY }}" | gpg --batch --passphrase "$PASSPHRASE" --import
+        env:
+          PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+
+      - name: "Deploy Locally"
+        run: >
+          mvn
+          $MVN_CLI_ARGS
+          -DaltDeploymentRepository=local::file:./temp_local_repo
+          -P release
+          deploy
+        env:
+          MAVEN_CENTRAL_USER: ${{ secrets.MAVEN_CENTRAL_USER }}
+          MAVEN_GPG_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+
+      - name: "Deploy Staging"
+        run: |
+          # Create an empty directory to correctly execute the plugin
+          mkdir temp
+          cd temp
+          MVN_ARGS="$MVN_CLI_ARGS org.sonatype.plugins:nexus-staging-maven-plugin:1.7.0:deploy-staged-repository -DserverId=ossrh -DnexusUrl=https://oss.sonatype.org -DrepositoryDirectory=../temp_local_repo -DstagingProfileId=$MAVEN_CENTRAL_PROFILE_ID"
```

Comment:
(Question)

`-DnexusUrl` is required? What does it do?

Reply:
Yes, you can release to different repositories

Reply:
It confused me that for server selection we actively use reserved argument `-DserverId` (resulting in `https://oss.sonatype.org` implicitly). While additionally we have custom(?) argument `-DnexusUrl` set to `https://oss.sonatype.org` explicitly. This looks like duplicate information.

But since we have custom plugin `nexus-staging-maven-plugin` executing `deploy-staged-repository`, I have to assume this argument is originating from that third party.